app/routes/endemics/date-of-visit-ms.js

```diff
@@ -45,254 +40,138 @@ const previousPageUrl = (latestVetVisitApplication, typeOfReview, previousClaims
-          dateOfVisit: {
-            day: {
-              value: request.payload['visit-date-day'],
-              error: error.details.find(
-                (e) =>
-                  e.context.label === 'visit-date-day' ||
-                e.type.startsWith('dateOfVisit')
-              )
-            },
-            month: {
-              value: request.payload['visit-date-month'],
-              error: error.details.find(
-                (e) =>
-                  e.context.label === 'visit-date-month' ||
-                e.type.startsWith('dateOfVisit')
-              )
-            },
-            year: {
-              value: request.payload['visit-date-year'],
-              error: error.details.find(
-                (e) =>
-                  e.context.label === 'visit-date-year' ||
-                e.type.startsWith('dateOfVisit')
-              )
-            },
-            errorMessage: error.details.find((e) =>
-              e.context.label.startsWith('visit-date')
-            )
-              ? {
-                  text: error.details.find((e) =>
-                    e.context.label.startsWith('visit-date')
-                  ).message
-                }
-              : undefined
-          },
-          backLink: previousPageUrl(latestVetVisitApplication, typeOfReview, previousClaims, typeOfLivestock)
+const getOldWorldClaimFromApplication = (oldWorldApp, typeOfLivestock) =>
+  oldWorldApp && typeOfLivestock === oldWorldApp.data.whichReview
+    ? {
+        statusId: oldWorldApp.statusId,
+        data: {
+          claimType: oldWorldApp.data.whichReview,
+          dateOfVisit: oldWorldApp.data.visitDate
         }
-      : {}
+      }
+    : undefined
+
+const getInputErrors = (request, reviewOrFollowUpText) => {
+  const inputtedValues = {
+    day: request.payload[labels.day],
+    month: request.payload[labels.month],
+    year: request.payload[labels.year]
+  }
+
+  const inputsInError = {
+    day: false,
+    month: false,
+    year: false
+  }
+
+  if (inputtedValues.day === '' || Number.isNaN(Number(inputtedValues.day)) || inputtedValues.day > 31) {
+    inputsInError.day = true
+  }
+
+  if (inputtedValues.month === '' || Number.isNaN(Number(inputtedValues.month)) || inputtedValues.month > 12) {
+    inputsInError.month = true
+  }
+
+  if (inputtedValues.year === '' || Number.isNaN(Number(inputtedValues.year))) {
+    inputsInError.year = true
+  }
+
+  const inputsInErrorResults = Object.entries(inputsInError)
+
+  const firstFoundInputError = inputsInErrorResults.find(([_inputName, inputInError]) => inputInError)
+
+  if (firstFoundInputError) {
+    const inputNameInError = firstFoundInputError[0]
+    return {
+      errorSummary: [{
+        text: 'Enter a date in the boxes below',
+        href: `#visit-date-${inputNameInError}`
+      }],
+      inputsInError
+    }
+  }
+
+  const dayInput = request.payload[labels.day].trim()
+  const dateSection = dayInput.length === 1 ? `0${dayInput}` : dayInput
```

Comment:
[padStart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart) has your back here:
```suggestion
  const dateSection = request.payload[labels.day].trim().padStart(2, '0')
```

Reply:
Didnt realise this was available on strings, ace!